### ISSUE TITLE
[Bug] Notifications causes lag and crashes

### ISSUE THREAD

**Issue by zDyanTB** [Jas-SinghFSU/HyprPanel#170]
**Describe the bug**
Notifications won't disappear and can cause stutters and crashes

**To Reproduce**
Enable notifications

**Expected behavior**
Appear/disappear properly

Video:
https://streamable.com/6l1tz1

 - Distribution: NixOs
 - Window Manager/Desktop Environment: Hyprland

**Comment by zDyanTB** [comment]
It seems the whole bar is laggy 
![image](https://github.com/user-attachments/assets/fee02946-163c-4d2f-a667-e84d60da86b2)
actual time: 01:10 PM

**Comment by Jas-SinghFSU** [comment]
Is this a new issue for you or has it always occurred?

**Comment by zDyanTB** [comment]
New, since it updated it

**Comment by Jas-SinghFSU** [comment]
Do you happen to know what your last updated version/commit was? I wonder if it's still an issue in the old version.

**Comment by zDyanTB** [comment]
I have it on Flakes
"lastModified": 1723433930,
"rev": "79bf05d9c99d04f4e93395de8d6477c8473eacaa",

The new one:
"lastModified": 1724309181,
"rev": "ff29790317003d4071205b303ce2ccc149f8cd21",

**Comment by zDyanTB** [comment]
note: `hyprpanel --clear-cache` solves it, but my config is gone..

**Comment by Jas-SinghFSU** [comment]
Are you referring to this command?
```
ags --clear-cache
```

**Comment by zDyanTB** [comment]
```console
❯ hyprpanel -h
USAGE:
    /nix/store/0yw867srsmmjzlzbyrd3shjbabrp7jas-ags-1.8.2/bin/.ags-wrapped [OPTIONS]
    
OPTIONS:
......
    --clear-cache           Remove /home/zdyant/.cache/ags and exit
    ```

**Comment by Jas-SinghFSU** [comment]
If so then it delete the directory where your config is stored that persists between sessions (hence the cache). I wonder if AGS's `Preserve Actions` is causing this behavior. 

I'll paste what I wrote in the other command and maybe it's applicable here as well:

> If you are able to still replicate this, could you - once replicated - try deleting your ~/.cache/ags/notifications/ directory and disabling Preserve Actions in Settings > Configuration > Notifications?

**Comment by Jas-SinghFSU** [comment]
If that is indeed the culprit then I'll have to disable that feature by default and provide a warning against it.

**Comment by Jas-SinghFSU** [comment]
> ```
> ❯ hyprpanel -h
> USAGE:
>     /nix/store/0yw867srsmmjzlzbyrd3shjbabrp7jas-ags-1.8.2/bin/.ags-wrapped [OPTIONS]
>     
> OPTIONS:
> ......
>     --clear-cache           Remove /home/zdyant/.cache/ags and exit
>     ```
> ```

Ah yea, on nix `hyprpanel` is essentially running `ags` commands so that make sense.

**Comment by Jas-SinghFSU** [comment]
After further discussion in discord, it seems that this issue has been resolved by #199 for now.
